libraries/plugins/peerplays_sidechain/sidechain_net_handler_bitcoin.cpp

```diff
@@ -664,109 +867,80 @@ void sidechain_net_handler_bitcoin::recreate_primary_wallet() {
                std::string active_pw_address = active_pw_pt.get_child("result").get<std::string>("address");
                std::string prev_pw_address = prev_sw_pt.get<std::string>("address");
 
-               transfer_all_btc(prev_pw_address, active_pw_address);
+               if (prev_pw_address == active_pw_address) {
+                  elog("BTC previous and new primary wallet addresses are same. No funds moving needed [from ${prev_sw} to ${active_sw}]", ("prev_sw", prev_sw->id)("active_sw", active_sw->id));
+                  return;
+               }
+
+               uint64_t fee_rate = bitcoin_client->estimatesmartfee();
+               uint64_t min_fee_rate = 1000;
+               fee_rate = std::max(fee_rate, min_fee_rate);
+
+               double min_amount = ((double)fee_rate / 100000000.0); // Account only for relay fee for now
+               double total_amount = 0.0;
+               std::vector<btc_txout> inputs = bitcoin_client->listunspent_by_address_and_amount(prev_pw_address, 0);
+
+               if (inputs.size() == 0) {
+                  elog("Failed to find UTXOs to spend for ${pw}", ("pw", prev_pw_address));
+                  return;
+               } else {
+                  for (const auto &utx : inputs) {
+                     total_amount += utx.amount_;
+                  }
+
+                  if (min_amount >= total_amount) {
+                     elog("Failed not enough BTC to transfer from ${fa}", ("fa", prev_pw_address));
+                     return;
+                  }
+               }
+
+               fc::flat_map<std::string, double> outputs;
+               outputs[active_pw_address] = total_amount - min_amount;
+
+               std::string tx_str = create_transaction(inputs, outputs);
+
+               if (!tx_str.empty()) {
+
+                  auto active_sons = gpo.active_sons;
+                  std::vector<son_id_type> signers;
+                  for (const son_info &si : active_sons) {
+                     signers.push_back(si.son_id);
+                  }
```

Comment:
We should use a set of active sons for previous wallet,  not for the new one to sign transfer transaction (it is transaction from the old wallet to the new wallet,  so the old wallet owners should sign it). What is stored in the `gpo.active_sons` at this moment?

Reply:
Good catch. I'll fix it asap.